packages/core/src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -384,20 +384,24 @@ export async function uploadArtifactToS3(
         headersObj['x-amz-server-side-encryption-aws-kms-key-id'] = resp.kmsKeyArn
     }
 
+    let requestId: string | undefined = undefined
+    let id2: string | undefined = undefined
+    let responseCode: string = ''
+
     try {
         const response = await request.fetch('PUT', resp.uploadUrl, {
             body: readFileSync(fileName),
             headers: resp?.requestHeaders ?? headersObj,
         }).response
         logger.debug(`StatusCode: ${response.status}, Text: ${response.statusText}`)
+        requestId = response.headers?.get('x-amz-request-id') ?? undefined
+        id2 = response.headers?.get('x-amz-id-2') ?? undefined
+        responseCode = response.status.toString()
```

Comment:
Not great that all of this is placed in the business logic for a feature. This kind of thing should live in `shared/request`.